test/notebook_tests/test_notebooks.py

```diff
@@ -0,0 +1,95 @@
+import fileinput
+import os
+import logging
+import nbformat
+import pytest
+import traceback
+
+from shutil import copyfile
+from nbconvert.preprocessors import ExecutePreprocessor
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+_EXCLUSIVE_DEVICE_REGIONS = {
+    "TN1": {
+        "ARN": "arn:aws:braket:::device/quantum-simulator/amazon/tn1",
+        "REGION": [
+            "us-west-2",
+            "us-east-1",
+        ],
+    }
+}
+
+demo_path = "examples/"
+demo_notebooks = []
+
+
+for dir_, _, files in os.walk(demo_path):
+    for file_name in files:
+        rel_file = os.path.join(dir_, file_name)
+        if file_name.endswith(".ipynb") and "QPU" not in file_name:
+            demo_notebooks.append((dir_, rel_file))
+
+
+def _rename_bucket(notebook_path, s3_bucket):
+    with fileinput.FileInput(notebook_path, inplace=True) as file:
+        for line in file:
+            print(
+                line.replace("amazon-braket-Your-Bucket-Name", s3_bucket).replace(
+                    "amazon-braket-<bucket name>", s3_bucket
+                ),
+                end="",
+            )
+
+
+def _check_exclusive_device_availability(notebook_path, region):
+    device_arn = _EXCLUSIVE_DEVICE_REGIONS["TN1"]["ARN"]
+    if region not in _EXCLUSIVE_DEVICE_REGIONS["TN1"]["REGION"]:
```

Comment:
Instead of referencing `TN1` directly, loop over `_EXCLUSIVE_DEVICE_REGIONS`

Reply:
Thanks, have made the suggested updates.